services/QuillLMS/client/app/bundles/Connect/components/questions/responseComponent.jsx

```diff
@@ -439,11 +440,11 @@ class ResponseComponent extends React.Component {
 
   renderRematchAllButton = () => {
     const { filters } = this.props
-    let disabled = filters.numberOfResponses > 1000
+
     if (this.props.admin) {
       const text = this.state.progress ? `${this.state.progress}%` : 'Rematch Responses';
 
-      return (<button className="button is-outlined is-danger" disabled={disabled} onClick={this.rematchAllResponses} style={{ float: 'right', }} type="button">{text}</button>);
+      return (<button className="button is-outlined is-danger" disabled={!this.state.enableRematchAllButton} onClick={this.setState({enableRematchAllButton: false}) && this.rematchAllResponses} style={{ float: 'right', }} type="button">{text}</button>);
```

Comment:
I would just put `this.setState({ enableRematchAllButton: false })` as the first line in `rematchAllResponses`, since this is the only place that gets called anyway.

Reply:
That's fair, that'll look a little cleaner. Thanks for the suggestion, will do!